src/Microsoft.OData.Core/ODataInnerError.cs

```diff
@@ -93,5 +133,34 @@ internal string ToJson()
                 this.StackTrace == null ? "" : JsonValueUtils.GetEscapedJsonString(this.StackTrace),
                 this.InnerError == null ? "{}" : this.InnerError.ToJson());
         }
-}
+
+        private string LookUpStringValue(string v)
+        {
+            if (Properties == null)
+            {
+                return string.Empty;
+            }
+
+            foreach (KeyValuePair<string, ODataValue> pair in Properties)
+            {
+                if (pair.Key == v)
+                {
+                    string value = (string)pair.Value.FromODataValue();

```

Comment:
>(string) [](start = 35, length = 8)

static cast to string maybe unsafe. Shall we test it using "ODataPrimitiveValue" with String type?

Reply:
Or call "ToString()"?

---
In reply to: [320874750](https://github.com/OData/odata.net/pull/1523#discussion_r320874750) [](ancestors = 320874750)